src/effects/lv2/LV2Effect.cpp

```diff
@@ -1408,8 +1409,9 @@ size_t LV2Effect::RealtimeProcess(int group, float **inbuf, float **outbuf, size
    return numSamples;
 }
 
-bool LV2Effect::RealtimeProcessEnd()
+bool LV2Effect::RealtimeProcessEnd() noexcept
 {
+return GuardedCall<bool>([&]{
```

Comment:
`R GuardedCall(const F1 &body, const F2 &handler = F2::Default(), std::function<void(AudacityException*)> delayedHandler = DefaultDelayedHandlerAction{})`
Don't you think that the last parameter makes this call not completely exception-safe?

Reply:
Do you mean that delayedHander should be noexcept?  But as the name suggests, its execution is delayed, and always in the main thread, invoked from the event loop, where exceptions escaping from it would also be caught by the top-level exception handler.  It is not invoked in the catch block.

The second argument is the non-delayed handler and the documentation is explicit that this function may re-throw its argument, or throw another exception.

Reply:
Not exactly, `LV2Effect::RealtimeProcessEnd()` is now `noexcept` but I think that `std::function` constructor does not provide such guarantees

Reply:
I see what you mean.

From https://en.cppreference.com/w/cpp/utility/functional/function/function:  "5,10) Does not throw if f is a function pointer or a std::reference_wrapper, otherwise may throw std::bad_alloc or any exception thrown by the copy constructor of the stored callable object."

So, please see the added commit, where I make some effort to avoid potentially throwing construction of std::function.

I also add a lot of `noexcept` in AudacityException.h.  The proof that exceptions really don't happen is still not complete, though, without diving into `AudacityException::EnqueueAction` and from there into `BasicUI::CallAfter` to be sure no other kinds of exceptions escape besides `std::bad_alloc`.

Still there is an opinion of some that you should implement `https://en.cppreference.com/w/cpp/memory/new/set_new_handler` if you have a way to do it, and then ignore `std::bad_alloc`, because memory exhaustion is an unrecoverable situation which is going to lead to a `std::terminate` anyway.